main.tf

```diff
@@ -490,6 +510,52 @@ resource "kubernetes_secret" "postgresql_mtls_secret" {
   depends_on = [kubernetes_namespace.gitlab_namespace]
 }
 
+# Recover the SMTP password from the GCP secret Manager
+data "google_secret_manager_secret_version" "secret_managed_smtp_predefined_pass" {
+  secret    = var.gcp_existing_smtp_secret_name
+  project   = var.project_id
+  count     = var.gitlab_enable_smtp ? 1 : 0
+
+  depends_on = [kubernetes_namespace.gitlab_namespace]
+}
+
+# Secret for Smtp account 
+resource "kubernetes_secret" "gitlab_smtp_secret" {
+  metadata {
+    name      = "gitlab-smtp-secret"
+    namespace = var.gitlab_namespace
+  }
+
+  data  = {
+    password = data.google_secret_manager_secret_version.secret_managed_smtp_predefined_pass[0].secret_data
+  }
+
+  count      = var.gitlab_enable_smtp ? 1 : 0
+  depends_on = [kubernetes_namespace.gitlab_namespace]
+}
```

Comment:
the ideal would be that the module is in charge to create the kubernetes_secret too